### DISCUSSION TITLE
Size Mismatch issue when running MLP with WebKB dataset from pytorch geometric

### DISCUSSION THREAD

**Discussion by AmitRoy7781** [pyg-team/pytorch_geometric#2926] [Q&A]
I tried to run MLP on WebKB datasets (i.e. cornell, texas, wisconsin) loading from the pytorch geometric using this [tutorial](https://uvadlc-notebooks.readthedocs.io/en/latest/tutorial_notebooks/tutorial7/GNN_overview.html). But, those datasets follow a 10-fold cross validation split (train_mask, test_mask, val_mask has a shape of (# nodes, 10))  which gives a size mismatch error. When I tried to run MLP on citation datasets (i.g. cora, citeser) they work fine but datasets with cross validation creates the size mismatch issue problem. 

It would be really helpful if you kindly give me a quick answer @rusty1s

**Comment by rusty1s** [comment]
Sorry for the late reply, I was offline for a few days.

For a single training experiment, you have to pick a single `mask` out of the 10 provided splits, *e.g.*:
```python
train_mask = data.train_mask[:, 0]
```
This let's you train the model just like on the `Planetoid` citation datasets.

In order to train sequentially on all provided splits, simply wrap everything into a for-loop, *e.g.*:
```
for i in range(10):
    model = Model(...)
    optimizer = ...
    train_mask = data.train_mask[:, i]
    for epoch in range(epochs):
        train(model, optimizer, data, train_mask)
```

**Reply by AmitRoy7781** [reply]
yeah, I tried this and it worked. Thanks a lot @rusty1s
